.changeset/fuzzy-lizards-end.md

```diff
@@ -0,0 +1,34 @@
+---
+'@solana/react': minor
+---
+
+Added a `useSignIn` hook that, given a `UiWallet` or `UiWalletAccount`, returns a function that you can call to trigger a wallet's [&lsquo;Sign In With Solana&rsquo;](https://phantom.app/learn/developers/sign-in-with-solana) feature.
+
+#### Example
+
+```tsx
+import { useSignIn } from '@solana/react';
+
+function SignInButton({ wallet }) {
+    const csrfToken = useCsrfToken();
+    const signIn = useSignIn(wallet);
+    return (
+        <button
+            onClick={async () => {
+                try {
+                    const { account, signedMessage, signature } = await signIn({
+                        domain: window.location.host,
+                        nonce: csrfToken,
```

Comment:
Aside: It's extremely dumb that SIWE has `request-id` and `nonce` which are duplicative in almost all cases, and JWT-esque `issued-at` and `expires-at` timestamps when these are not intended to be and should not be used as JWTs directly. Apps should generate a random session ID from the server and use it as `request-id`, then verify and issue a proper JWT if needed from the server the correct way.

Ideally we can reflect the best way to do this in docs and examples and users should not touch most of the API.

Reply:
nit: I do prefer `requestId` to `nonce` because it makes it more clear that it should come from a server, it's not using a cryptography term improperly, and it's not confused by Brits for... other meanings of the term.